### PR TITLE
[core] d8-cni-configuration migration

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
## Description

Migration from d8-cni-configuration secret to proper cni module configs.
## Why do we need it, and what problem does it solve?

This migration simplifies cni autoconfiguration process and do it more clear to user.
## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: core
type: chore
summary: Migration from `d8-cni-configuration` secret to proper cni module configs.
impact: none of the cni pods should restart ↓
impact_level: high
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by RomanenkoDenys** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10371028697 10371028697-1-con-1-27 /sys/deckhouse-oss/install:pr9347 redos@89.169.131.252

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by RomanenkoDenys** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10375560692 10375560692-1-con-1-27 /sys/deckhouse-oss/install:pr9347 redos@89.169.134.27

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/static'...

**Review by name212 — APPROVED** [review]

File: modules/021-cni-cilium/hooks/set_cilium_mode.go [hunk 1]
```diff
@@ -17,81 +17,19 @@ limitations under the License.
...
-				input.Values.Set("cniCilium.internal.mode", ciliumConfig.Mode)
-			}
-			if ciliumConfig.MasqueradeMode != "" {
-				input.Values.Set("cniCilium.internal.masqueradeMode", ciliumConfig.MasqueradeMode)
-			}
-			return nil
-		}
+	value, ok := input.ConfigValues.GetOk("cniCilium.masqueradeMode")
+	if ok {
+		input.Values.Set("cniCilium.internal.masqueradeMode", value.String())
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/set_cilium_mode.go:32]: Get rid of internal version of masqueradeMode pls.

File: modules/021-cni-cilium/hooks/set_cilium_mode_test.go [hunk 2]
```diff
@@ -1,255 +0,0 @@
-/*
```
Comment (apolovov) [inline modules/021-cni-cilium/hooks/set_cilium_mode_test.go:1]: Don't delete the test, need to test simple conditions.

File: modules/021-cni-cilium/openapi/config-values.yaml [hunk 3]
```diff
@@ -22,6 +22,15 @@ properties:
       Create routes to Pods on other Nodes.
 
       All Nodes must be located in the same L2 domain.
+  masqueradeMode:
+    type: string
+    enum: ["Netfilter", "BPF"]
+    default: "BPF"
+    description: |
+      Cilium masquerade work mode.
```
Comment (apolovov) [inline modules/021-cni-cilium/openapi/config-values.yaml:30]: TODO: @apolovov think what to write.

File: modules/035-cni-flannel/hooks/set_pod_network_mode_test.go [hunk 4]
```diff
@@ -1,137 +0,0 @@
-/*
```
Comment (apolovov) [inline modules/035-cni-flannel/hooks/set_pod_network_mode_test.go:1]: don't delete the test, modify it.

File: global-hooks/migrate/migrate_d8_cni_secret.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+		case "Direct":
+			ciliumSettings["tunnelMode"] = "Disabled"
+		case "DirectWithNodeRoutes":
+			ciliumSettings["tunnelMode"] = "Disabled"
+			ciliumSettings["createNodeRoutes"] = true
+		default:
+			return fmt.Errorf("unknown cilium mode %s", ciliumConfig.Mode)
+		}
+
+		ciliumSettings["masqueradeMode"] = ciliumConfig.MasqueradeMode
```
Comment (apolovov) [inline global-hooks/migrate/migrate_d8_cni_secret.go:179]: Add a condition if ciliumConfig.MasqueradeMode is "".

Comment (apolovov) [reply global-hooks/migrate/migrate_d8_cni_secret.go:179]: Or above, check if the secret.cilium struct is empty.

File: global-hooks/migrate/migrate_d8_cni_secret.go [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+			Enabled:  pointer.Bool(true),
+			Settings: ciliumSettings,
+		}
+		cniModuleConfig.Spec.Version = 1
+
+	default:
+		return fmt.Errorf("unknown cni name: %s", cniName)
+	}
+
+	// create secret
```
Comment (apolovov) [inline global-hooks/migrate/migrate_d8_cni_secret.go:191]:
```suggestion
	// create ModuleConfig
```

**Review by apolovov — CHANGES_REQUESTED** [review]

File: modules/021-cni-cilium/hooks/set_cilium_mode.go (hunk 1 above)
Comment (RomanenkoDenys) [reply modules/021-cni-cilium/hooks/set_cilium_mode.go:32]: removed, thanks

File: modules/021-cni-cilium/hooks/set_cilium_mode_test.go (hunk 2 above)
Comment (RomanenkoDenys) [reply modules/021-cni-cilium/hooks/set_cilium_mode_test.go:1]: restored and modified

File: modules/035-cni-flannel/hooks/set_pod_network_mode_test.go (hunk 4 above)
Comment (RomanenkoDenys) [reply modules/035-cni-flannel/hooks/set_pod_network_mode_test.go:1]: restored and modified

File: global-hooks/migrate/migrate_d8_cni_secret.go (hunk 5 above)
Comment (RomanenkoDenys) [reply global-hooks/migrate/migrate_d8_cni_secret.go:179]: done

**Review by apolovov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/openstack'...

File: .golangci.yaml [hunk 7]
```diff
@@ -44,4 +44,4 @@ linters:
   - unconvert
   - unparam
   - whitespace
-  - copyloopvar
+#  - copyloopvar
```
Comment (z9r5) [inline .golangci.yaml:47]:
```suggestion
```

**Pull request merged**